hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/TypedTable.java

```diff
@@ -94,25 +95,33 @@ public TypedTable(RDBTable rawTable,
       CodecRegistry codecRegistry, Class<KEY> keyType,
       Class<VALUE> valueType,
       CacheType cacheType, String threadNamePrefix) throws IOException {
-    this.rawTable = Objects.requireNonNull(rawTable, "rawTable==null");
-    Objects.requireNonNull(codecRegistry, "codecRegistry == null");
-
-    this.keyType = Objects.requireNonNull(keyType, "keyType == null");
-    this.keyCodec = codecRegistry.getCodecFromClass(keyType);
-    Objects.requireNonNull(keyCodec, "keyCodec == null");
-
-    this.valueType = Objects.requireNonNull(valueType, "valueType == null");
-    this.valueCodec = codecRegistry.getCodecFromClass(valueType);
-    Objects.requireNonNull(valueCodec, "valueCodec == null");
-
-    this.supportCodecBuffer = keyCodec.supportCodecBuffer()
```

Comment:
These null checks are bypassed, if keyType/valuetype is null it will throw an IllegalStateException . We can keep these checks.
Objects.requireNonNull(keyType, "keyType == null");
Objects.requireNonNull(valueType, "valueType == null");

Reply:
We can not put above check before being used as java restriction,

public TypedTable(RDBTable rawTable,
      CodecRegistry codecRegistry, Class<KEY> keyType,
      Class<VALUE> valueType,
      CacheType cacheType, String threadNamePrefix) throws IOException {

> **<---- before this(), its not allowed to add above check, and after this() has no meaning of this check.**

    this(rawTable, codecRegistry.getCodecFromClass(keyType), codecRegistry.getCodecFromClass(valueType),
        cacheType, threadNamePrefix);
  }

Reply:
can do this
```java
this(rawTable, codecRegistry.getCodecFromClass(Objects.requireNonNull(keyType,"keyType == null"), codecRegistry.getCodecFromClass(Objects.requireNonNull(valueType,"valueType == null")), threadNamePrefix);
```

Reply:
Or , move the check to `CodecRegistry`.
```java
@ -61,6 +62,7 @@ private CodecMap(Map<Class<?>, Codec<?>> map) {
     }
 
     <T> Codec<T> get(Class<T> clazz) {
+      Objects.requireNonNull("clazz == null");
       final Codec<?> codec = map.get(clazz);
       return (Codec<T>) codec;
     }
```